react/src/components/organisms/GenTeaser/TeaserSearch/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { HeaderSearch } from '../../../../index';
+
+class TeaserSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: ''
+    };
+  }
+
+  onChange = (term) => {
+    this.setState({
+      query: term
+    });
+  }
+
+  onClick = (e) => {
+    e.preventDefault();
+    const { target, queryInput } = this.props;
+    const { query } = this.state;
+    if (query.length > 0) {
+      const searchURL = queryInput ? target.replace(`{${queryInput}}`, this.state.query) : target;
+      this.redirect(searchURL);
+    }
+  }
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    const { target, queryInput } = this.props;
+    const { query } = this.state;
+    if (query.length > 0) {
+      const searchURL = queryInput ? target.replace(`{${queryInput}}`, this.state.query) : target;
+      this.redirect(searchURL);
+    }
+  }
+
+  redirect = (searchURL) => {
+    if (window.location !== window.parent.location) {
+      window.parent.location.assign(searchURL);
+    } else {
+      window.location.assign(searchURL);
+    }
+  }
+
+  render() {
+    const { placeholder, id, ...rest } = this.props;
+    return(
+      <HeaderSearch
+        buttonSearch={{
+          'aria-label': '',
+          onClick: (e) => this.onClick(e),
+          text: 'Search',
+          usage: ''
+        }}
+        defaultText=""
```

Comment:
I wonder if we can just pass queryInput into defaultText as this prop is used and tracked similarly in HeaderSearch

Reply:
updated in 506913d and remove unnecessary state tracking